### PR TITLE
refactor/correctness: Move `GroupInfo` into own module and tighten-up serialization.

### PR DESCRIPTION
**Pull request by duesee**
Over the course of preparing a short blog post about validation in OpenMLS I cleaned up `GroupInfo` a bit.

This PR does the following:

* Move `GroupInfo` (and all related types) into a new `group_info` module.
* Make it so that `VerifiableGroupInfo` does not (unnecessarily) derive `Serializable`. `Serializable` is test-gated now.
* `#[derive(TlsSerialize, TlsSize)]` for `GroupInfo` instead of implementing it.
* Remove unused `test_pgs.rs`.

### CODE REVIEW

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks, generally looks good two me with a nit and a question.

Do we really want to delete `test_pgs.rs`? The names etc. aren't right anymore but did you check that the tests aren't useful?

File: openmls/src/framing/message_in.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@
 //!
 //! The [`MlsMessageIn`] struct is meant to be deserialized upon receiving it
 //! from the DS. After deserialization, its content (either a [`PublicMessage`],
-//! [`PrivateMessage`], [`KeyPackage`], [`Welcome`] or [`GroupInfo`]) can be
+//! [`PrivateMessage`], [`KeyPackage`], [`Welcome`] or [`crate::messages::group_info::GroupInfo`]) can be
```
[the remark this answers is not in the document]
Comment (duesee) [reply openmls/src/framing/message_in.rs:10]: That's nicer, yeah :+1:

**Comment by duesee** [comment]
`test_pgs.rs` is an empty file. Probably a leftover from another PR or merge. But let me take that as an opportunity to re-check if we really test what we want.

File: openmls/src/framing/message_in.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (duesee) [reply openmls/src/framing/message_in.rs]: Done in 821f20b76c73b2a5e9f937221041cac261120d52

**Comment by duesee** [comment]
> But let me take that as an opportunity to re-check if we really test what we want.

The test were moved into `test_export_group_info.rs` in https://github.com/openmls/openmls/commit/b56883aaec03a3c64983b6ea44576530785554d9#diff-9cae41c82a5e5d4f72878ed55cf19a4580d42b6b3e87a2a2a1eb1701100e82ff. I also think they suffice for now (export + verify signature).

**Comment by duesee** [comment]
Commited some unrelated stuff by accident and reverted.

**Review by franziskuskiefer — APPROVED** [review]
Thanks

**Pull request merged**